### PR TITLE
Installer improvements

### LINKED ISSUE #142: SSH-ing into Jenkins container fails
When I pick scenario 3 (Dockerized jenkins and Gitlab), install fails repeatedly at this point:

```
Granting permissions to other users to pip install....Completed
['18.222.73.235:8081', 'admin', '8dabee45bb234a96bcd2a7c594acf15f', '18.222.73.235', 'root', '2200', 'sg-852143ee', 'subnet-aa5d53d2']
Unable to SSH into the Jenkins instance! Is the jenkinskey key or the username valid? Jenkins IP: 18.222.73.235 Username: root Keypath: /home/centos/jazz-installer/installscripts/dockerfiles/jenkins//jenkinskey.pem Port: 2200
```

Originally this just logged the failure and did not stop script execution, I added the `sys.exit` since there's no point in continuing if this fails. I also added some logging to see why the SSH call (copied below)using paramiko fails

```
def check_jenkins_sshuser_valid(parameter_list, port_number, keypath):
    """
        Check if the ssh login name is a user
    """
    jenkins_server_public_ip = parameter_list[3]
    jenkins_server_ssh_login = parameter_list[4]
    keyfile = keypath + "/jenkinskey.pem"
    try:
        ssh = paramiko.SSHClient()
        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        ssh.connect(jenkins_server_public_ip, username=jenkins_server_ssh_login, key_filename=keyfile, port=port_number)
    except:
        sys.exit("Unable to SSH into the Jenkins instance! Is the jenkinskey key or the username valid? Jenkins IP: %s Username: %s Keypath: %s Port: %s" %(jenkins_server_public_ip, jenkins_server_ssh_login, keyfile, port_number,))
```

The ssh key is in the correct place, but I can verify that the paramikio ssh failed. Looking in to this. 

When dealing with containers we shouldn't even need to SSH in in most cases, it's a bit roundabout, we can just `docker exec` and skip the part where we have to copy ssh keys around.

### PR DESCRIPTION
**Pull request by bleggett**
More fixes to decouple the wizard from the Terraform scripts in service of #110 

- Remove some unnecessary uses of `sudo`: Don't wanna escalate to root unless we need to.

-  Skip download and install of some tools if they are already installed: This was for my own sanity, to speed up repeat runs of the script

- It was impossible to install using Terraform directly because the wizard was 'reaching past' Terraform and updating some Chef variables, removed that. Terraform now updates the Chef playbook variables itself before copying the playbook over. This means you can stand up the entire stack with Terraform alone if you gather and set the correct input variables.

- Introduced `terraform.tfvars`: Rather than set Terraform variables by setting the default values in the variable definition files to the desired values, do things "the Terraform way" and keep variable definitions and variable assignments separate. This has the beneficial side effect of isolating all the Terraform variables that the wizard sets into one spot (`terraform.tfvars`). **The wizard has essentially one job now: Collect user input and update `terraform.tfvars` with that input, and invoke Terraform.**

- Reorg/cleanup/lint/format the wizard Python scripts a bit.

 - Fix #142 by removing the cosmetic SSH check.

There is more work that needs to be done, we should probably create separate Terraform modules for each scenario, we overuse `sed` especially in the Python scripts, etc etc. But this is a start.

I have verified this on a clean CentOS box in AWS using scenario 3 (up until I got to the point where the installer fails because I'm working out of a fork and it hardcodes repo/branch names). Because I wasn't able to test the other scenarios and I'm a pessimist I bet they'll break.

### CODE REVIEW

File: installscripts/Installer.sh [hunk 1]
```diff
@@ -133,24 +153,24 @@ function install_packages () {
...
+      # Download and Install awscli
+      sudo pip install awscli >> $LOG_FILE &
+      spin_wheel $! "Downloading & installing awscli bundle"
+  fi
 
-  #Undo output redirection
-  exec 1> /dev/stdout
-  exec 2> /dev/stderr
+  #Undo output redirection and close unused file descriptors.
+  exec 1>&3 3>&-
```
Comment (suryajak) [inline installscripts/Installer.sh]: Thanks for fixing this, we were seeing symptoms of this in our daily stack where the jenkins box after a few builds will talk about one too many open file handles. We never got to understand the issue, and this makes me think its related.

File: installscripts/wizard/scenarios/jazz_generic_details.py [hunk 2]
```diff
@@ -66,18 +75,21 @@ def get_stack_generic_details(jazz_branch):
 
     region = raw_input("AWS Region (e.g. us-east-1): ")
     if region not in knownWorkingRegions:
-        print('Warning: This installer has not been tested against the region you specified.\nPlease check the Jazz documentation (https://github.com/tmobile/jazz-installer/wiki#prerequisites) to verify the region you have chosen supports the required AWS resources.')
+        print(
+            'Warning: This installer has not been tested against the region you specified.\nPlease check the Jazz documentation (https://github.com/tmobile/jazz-installer/wiki#prerequisites) to verify the region you have chosen supports the required AWS resources.'
```
Comment (suryajak) [inline installscripts/wizard/scenarios/jazz_generic_details.py]: Can we also put a user prompt and have them confirm if they want to continue?

File: installscripts/wizard/scenarios/jazz_generic_details.py [hunk 3]
```diff
@@ -40,23 +42,30 @@ def set_aws_config(region):
...
+            "Please provide a prefix for your stack (limited to 13 characters)(eg: myjazz) :"
+        )
     tag_env_prefix = tag_env_prefix.lower()
 
-    #TODO - Need to check if we really need this
-    tag_enviornment="Development"
-    tag_exempt=(datetime.datetime.today()+datetime.timedelta(days=1)).strftime("%m/%d/%Y")
-    tag_owner=tag_env_prefix+"-Admin"
+    # TODO Since most of these are currently static we could define them with interpolation in envprefix.tf
+    tag_enviornment = "Development"
```
Comment (suryajak) [inline installscripts/wizard/scenarios/jazz_generic_details.py]: carry over typo tag_envir**o**nment

File: installscripts/Installer.sh (hunk 1 above)
Comment (bleggett) [reply installscripts/Installer.sh]: Ah, yes, it might be.

File: installscripts/wizard/scenarios/jazz_generic_details.py (hunk 2 above)
Comment (bleggett) [reply installscripts/wizard/scenarios/jazz_generic_details.py]: Yep.

File: installscripts/wizard/scenarios/jazz_generic_details.py (hunk 3 above)
Comment (bleggett) [reply installscripts/wizard/scenarios/jazz_generic_details.py]: Fixed

**Pull request merged**